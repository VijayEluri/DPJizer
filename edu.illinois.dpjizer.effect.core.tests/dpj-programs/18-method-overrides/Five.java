/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class A {
	region Rx, Ry;
	int x in Rx;
	int y in Ry;
	
	A() {}
	
	void m() {
		x = 1;
	}
}

class B extends A {
	B() {}
	
	@Override
	void m() {
		y = 2;
	}
}
