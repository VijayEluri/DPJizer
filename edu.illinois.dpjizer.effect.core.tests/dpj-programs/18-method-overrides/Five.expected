A.A() invokes java.lang.Object.Object()
A.m() writes Root : A.Rx
B.B() invokes A.A()
B.m() writes Root : A.Ry overrides A.m()